tui/installer/__init__.py

```diff
@@ -72,20 +75,20 @@ def preserve_settings(answers):
     def preserve_timezone(answers):
         if not_preserve_settings(answers):
             return False
-        if 'installation-to-overwrite' not in answers:
-            return False
-        settings = answers['installation-to-overwrite'].readSettings()
-        return 'timezone' in settings and 'request-timezone' not in settings
+        if 'installation-to-overwrite' in answers and answers['installation-to-overwrite'].settingsAvailable():
+            settings = answers['installation-to-overwrite'].readSettings()
+            return 'timezone' in settings and 'request-timezone' not in settings
+        return False
     not_preserve_timezone = lambda a: not preserve_timezone(a)
 
     def ha_enabled(answers):
         settings = {}
-        if 'installation-to-overwrite' in answers:
+        if 'installation-to-overwrite' in answers and answers['installation-to-overwrite'].settingsAvailable():
             settings = answers['installation-to-overwrite'].readSettings()
         return 'ha-armed' in settings and settings['ha-armed']
 
     def out_of_order_pool_upgrade_fn(answers):
-        if 'installation-to-overwrite' not in answers:
+        if not ('installation-to-overwrite' in answers and answers['installation-to-overwrite'].settingsAvailable()):
```

Comment:
ditto / likewise, and
the `key not in dict` pattern and together with `or not` is also easier or less difficult to read for me:
```suggestion
        if (
            "installation-to-overwrite" not in answers
            or not answers["installation-to-overwrite"].settingsAvailable()
        ):
```